meson.build

```diff
@@ -0,0 +1,117 @@
+project('cubic-server', 'cpp',
+  version : '0.1',
+  default_options : ['warning_level=3',
+                     'cpp_std=c++20'])
+
+#======================
+# variables declaration
+#======================
+
+cxxflags =          []
+ldflags  =          []
+cubic_include =     []
+cubic_src =         []
+cubic_deps =        []
+
+#=================================
+# use libc++ on clang if installed
+#=================================
+
+if get_option('libcpp_clang')
+    compiler = meson.get_compiler('cpp')
+    if compiler.get_id() == 'clang'
+        cxxflags += ['-stdlib=libc++']
+        ldflags +=  ['-stdlib=libc++']
+    else
+        warning('Clang only option ignored because used on compiler ' + compiler.get_id())
+    endif
+endif
+
+#=====================
+# set name and version
+#=====================
+
+cxxflags += ['-DPROGRAM_NAME="CubicServer"', '-DPROGRAM_VERSION="' + meson.project_version() + '"' ]
+
+#=====================
+# Manage dependencies.
+#=====================
+
+# pkgconfig is required for meson to work correctly.
+# dependencies might not be found without it on linux.
+
+thread = dependency('threads',   required: true)
+gtkmm =  dependency('gtkmm-3.0', required: true)
+curl =   dependency('libcurl', required: true)
+
+yaml     = dependency('yaml-cpp',      required: false)
+json     = dependency('nlohmann_json', required: false)
+noise    = dependency('perlin_noise',  required: false)
+argparse = dependency('argparse',      required: false)
+crcpp    = dependency('CRCpp',         required: false)
+
+if get_option('buildtype') == 'debug' and get_option('test')
+  gtest = dependency('gtest', main: true, required: false)
+  rapidcheck = dependency('rapidcheck', required: false)
+endif
```

Comment:
Since testing is already a build option, why not just always create the test target when tests are requested?

Reply:
Indeed, I did not know this was possible at the time, I will modify this accordingly